src/xclim/indices/generic.py

```diff
@@ -839,6 +846,42 @@ def season(
     return out
 
 
+def season_length_from_boundaries(season_start: xr.DataArray, season_end: xr.DataArray) -> xr.DataArray:
+    """
+    Season length using pre-computed boundaries.
+
+    Parameters
+    ----------
+    season_start : xr.DataArray
+        Day of year where the season starts.
+    season_end : xr.DataArray
+        Day of year where the season ends.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        Length of the season.
+
+    Notes
+    -----
+    If `season_start` and `season_end` are computed with different resampling frequencies, the time
+    of `season_start` are selected to write the output. This is useful when season start and end were computed
+    with different resampling frequencies. Otherwise, functions in ``xclim.indices.run_length`` will be appropriate.
```

Comment:
```suggestion
    The time of `season_start` is selected to write the output. This is only useful when season start and end were computed
    at an annual frequency but with different anchor months. Otherwise, functions in ``xclim.indices.run_length`` will be appropriate.
```
Could we add a check at the beginning of the function to ensure both inputs are annual and raise otherwise ?